pkg/gatewayserver/gatewayserver.go

```diff
@@ -642,50 +645,54 @@ func (gs *GatewayServer) startDisconnectOnChangeTask(conn connectionEntry) {
 		Context: conn.Context(),
 		ID:      fmt.Sprintf("disconnect_on_change_%s", unique.ID(conn.Context(), conn.Gateway().GetIds())),
 		Func: func(ctx context.Context) error {
-			d := random.Jitter(gs.config.FetchGatewayInterval, gs.config.FetchGatewayJitter)
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case <-time.After(d):
-			}
+			for {
```

Comment:
This change is basically looping indefinitely explicitly (via `for`), instead of implicitly (via `task.RestartAlways`).

The reason for this change is that `task.RestartAlways` will introduce jitter, even when no error occurs, due to the way `task.DialBackoffConfig` works. That jitter is based on a hardcoded sequence, which is not really modifiable in tests.

Hence, now the loop is explicit in order to have the retrieval interval depend only on `FetchGatewayInterval`.